packages/react/src/primitives/Pagination/PaginationItem.tsx

```diff
@@ -17,11 +17,20 @@ export const PaginationItem: React.FC<PaginationItemProps> = ({
   ariaLabel,
   ...rest
 }) => {
+  const onChange = React.useCallback(() => {
```

Comment:
The error here was that useCallback was being called inside of a conditional, so I've moved them out of the `switch`.